pkg/network/server.go

```diff
@@ -402,6 +414,19 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.BlockHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
+	if s.chain.HeaderHeight() < pong.LastBlockIndex {
+		return s.requestHeaders(p)
```

Comment:
I'm not sure it's enough, what if you're more than 2000 blocks behind? C# node is more correct here in that it uses a separate field for current known peer height and not just `Version().StartHeight`.

Reply:
added LastBlockIndex to Peer and init with `Version().StartHeight`